walrus_mainboard_driver/src/walrus_mainboard_driver/walrus_mainboard_driver.cpp

```diff
@@ -0,0 +1,670 @@
+		aei.registerHandle(effort_handle3);
+		
+		//Add diagnostics updaters
+		diagnostic_updater.setHardwareID("Walrus Main Board");
+		diagnostic_updater.add("Chassis", this, &MainBoardDriver::chassis_diagnostic_callback);
+		diagnostic_updater.add("Front Left Pod", boost::bind(&MainBoardDriver::pod_diagnostic_callback, this, _1, FL-1));
+		diagnostic_updater.add("Front Right Pod", boost::bind(&MainBoardDriver::pod_diagnostic_callback, this, _1, FR-1));
+		diagnostic_updater.add("Back Right Pod", boost::bind(&MainBoardDriver::pod_diagnostic_callback, this, _1, BR-1));
+		diagnostic_updater.add("Back Left Pod", boost::bind(&MainBoardDriver::pod_diagnostic_callback, this, _1, BL-1));
+		diagnostic_updater.add("Main Control Board", this, &MainBoardDriver::mainboard_diagnostic_callback);
+		diagnostic_updater.add("Battery 1", boost::bind(&MainBoardDriver::batt_diagnostic_callback, this, _1, 0));
+		diagnostic_updater.add("Battery 2", boost::bind(&MainBoardDriver::batt_diagnostic_callback, this, _1, 1));
+		diagnostic_updater.add("Battery 3", boost::bind(&MainBoardDriver::batt_diagnostic_callback, this, _1, 2));
+		diagnostic_updater.add("Battery 4", boost::bind(&MainBoardDriver::batt_diagnostic_callback, this, _1, 3));
+		diagnostic_updater.add("Power Systems", this, &MainBoardDriver::power_diagnostic_callback);
+		diagnostic_updater.add("Left Drive Motor Temperature", this, &MainBoardDriver::left_drive_temp_diagnostic_callback);
+		diagnostic_updater.add("Right Drive Motor Temperature", this, &MainBoardDriver::right_drive_temp_diagnsotic_callback);
+	}
+	
+	bool MainBoardDriver::init()
+	{	    
+	    //Setup publishers and subscribers to communicate with the embedded board
+		hs_control = nh.advertise<walrus_firmware_msgs::MainBoardHighSpeedControl>("/walrus/main_board/hs_control", 1000);
+		to_board = nh.advertise<walrus_firmware_msgs::MainBoardControl>("/walrus/main_board/PC_to_board_control", 1000);
+		hs_feedback = nh.subscribe("/walrus/main_board/hs_feedback", 1000, &MainBoardDriver::hs_feedback_callback, this);
+		ls_data = nh.subscribe("/walrus/main_board/ls_data", 1000, &MainBoardDriver::ls_data_callback, this);
+		from_board = nh.subscribe("/walrus/main_board/board_to_PC_control", 1000, &MainBoardDriver::from_board_callback, this);
+	    return true;
+	}
+	
+	void MainBoardDriver::read()
+	{
+        double temp, dr;
+	    double interval = (ros::Time::now() - last_read).toSec();
+	    last_read = ros::Time::now();
+	    
+	    {
+	        boost::lock_guard<boost::mutex> lock(control_data_mutex);
+	       
+	        for (int l = 0; l < 4; l++)
+	        {
+		        temp = (hs_feedback_msg.pod_position[l]/1024.0)*2*PI;
+		        if (POD_REV[l])
+		            temp = (2*PI) - temp;
+		        temp -= POD_POSITION_NEUTRAL[l];
+		        if (temp < 0)
+		            temp += 2*PI;
+		        dr = temp - pod_position[l];
+		        if (dr > 0 && dr > PI) //Decrease angle across 0
```

Comment:
I'm pretty sure this logic is wrong as you have a redundant case...
if dr > PI then it will also be greater than 0